pkg/await/states/job.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2016-2019, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package states
+
+import (
+	"fmt"
+
+	"github.com/pulumi/pulumi-kubernetes/pkg/logging"
+	"github.com/pulumi/pulumi/pkg/util/cmdutil"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func NewJobChecker() *StateChecker {
+	return &StateChecker{
+		conditions: []Condition{jobStarted, jobComplete},
+		readyMsg:   cmdutil.EmojiOr("✅ Job ready", "Job ready"),
+	}
+}
+
+//
+// Conditions
+//
+
+func jobStarted(obj metav1.Object) Result {
+	job := toJob(obj)
+	result := Result{Description: fmt.Sprintf("Waiting for Job %q to start", fqName(job))}
+
+	if job.Status.StartTime != nil {
+		result.Ok = true
+	}
+
+	return result
+}
+
+func jobComplete(obj metav1.Object) Result {
+	job := toJob(obj)
+	// TODO: may want to list active/succeeded/failed pods count
```

Comment:
+1 on this. We should do this for any resource that manages pods (Pods, Deploy, RS, StatefulSet etc)

Reply:
Updated to include this status:
```
[1/2] Waiting for Job "foo" to succeed (Active: 0 | Succeeded: 1 | Failed: 0)
```